dump_dbus_util.cpp

```diff
@@ -110,4 +97,99 @@ ManagedObjectType getDumpEntries(sdbusplus::bus::bus& bus)
     }
     return objects;
 }
+
+bool isSystemHMCManaged(sdbusplus::bus::bus& bus)
+{
+    using BiosBaseTableItem = std::pair<
+        std::string,
+        std::tuple<std::string, bool, std::string, std::string, std::string,
+                   std::variant<int64_t, std::string>,
+                   std::variant<int64_t, std::string>,
+                   std::vector<std::tuple<
+                       std::string, std::variant<int64_t, std::string>>>>>;
+    using BiosBaseTable = std::vector<BiosBaseTableItem>;
+    try
+    {
+        std::string hmcManaged{};
+        auto retVal = readDBusProperty<std::variant<BiosBaseTable>>(
+            bus, "xyz.openbmc_project.BIOSConfigManager",
+            "/xyz/openbmc_project/bios_config/manager",
+            "xyz.openbmc_project.BIOSConfig.Manager", "BaseBIOSTable");
+        const auto baseBiosTable = std::get_if<BiosBaseTable>(&retVal);
+        if (baseBiosTable == nullptr)
+        {
+            std::string err =
+                "Failed to read BIOSconfig property BaseBIOSTable";
+            log<level::ERR>(err.c_str());
+            throw std::runtime_error(err);
```

Comment:
Continuing the discussion here from https://github.com/ibm-openbmc/powervm-handler/pull/3#discussion_r807515864

> > Do we need to assume some default behavior i.e non-HMC managed system if there is a failure while reading the D-Bus property like added in other places in this function?

> good point, i want dhruv to add dependency on the BiosConfig manager service, this service should kick off after BiosConfig Manager has started. If not this exception will be thrown. But if the service is started and the attribute is not set to hmc managed then we can return false. Please revert back if you do not agree.

Looks, you are trying to get the BiosBaseTable type from the variant that you passed and If not, you are throwing the error does that capture anywhere (I guess, this is a utils function so it may be called anywhere) or the application will get crash?

and even I am wondering how that (service file changes) solves what you are trying to solve, the service dependency only takes care of the app starts not the above case, right?

If you added to handle the same case then, IMO nothing to do with this app right? and also, I am thinking about how this case occurs since it may be possible if the BiosBaseTable property signature changed and that is also handled by the readDBusProperty exception handler (read() call), right?

I think we are discussing some rare cases but, it occurs, I am trying to see what the expected behavior

Reply:
"BiosBaseTable property signature changed" this cannot be addressed, if property signature changes we should change every where.

I can remove throwing exception and by default assume it is non-hmc managed system if the property is not set, to be on safer side. Initially my thought was if we add "After=xyz.openbmc_project.biosconfig_manager.service" we are sure that the property will be set.